### ISSUE TITLE
Double lines because of paging issue

### ISSUE THREAD

**Issue by alexanderstautner** [crealytics/spark-google-analytics#16]
Hi,
I think I found a bug in your AnalyticsRelation.scala file.

The problem is in this excerpt of your code
```
  val maxPageSize = 10000
    val filtersString = combineFilters(filters)

    def queryDateRange(startDate: String, endDate: String) = {
      val queryWithoutFilter = analytics.data().ga()
        .get(ids, startDate, endDate, requiredMetrics.map("ga:" + _.name).mkString(","))
        .setDimensions(requiredDimensions.map("ga:" + _.name).mkString(","))
        .setMaxResults(maxPageSize)
      val query = if (filters.length > 0) queryWithoutFilter.setFilters(filtersString) else queryWithoutFilter
      val firstResult = query.execute
      val requiredPages = firstResult.getTotalResults / maxPageSize
      val restResults = (1 to requiredPages).flatMap { pageNum =>
        retry(3)(query.setStartIndex(pageNum * maxPageSize).execute.getRows.asScala).get
      }

```
When there are more than 10000 results, than you'll storage the results from 1 to 10000 in firstResult. 

restResults starts with result 10000 and not with result 10001. So there will be a duplication of the 10000th result  in the final result set.

Would be cool, if this problem would be fixed in the next version of this package.

**Comment by nightscape** [comment]
Hi @alexanderstautner,

did you observe that it downloads duplicate rows or did you just check the code?
The question if there are duplicates depends on whether the `setStartIndex` method is 0- or 1-based, right?

**Comment by alexanderstautner** [comment]
We observed a duplication of the line 10000 using your lib within pyspark. When we tried to insert the result of a spark dataframe via a pandas dataframe in a database, the database rejected the insertion with a key duplication error. 

Then we dumped the pyspark dataframe to a csv to explore the problem. We found that there was a duplication of result 10000.

We tried different dimension metics combinations and the error always occurs, when there were more than 10000th results returned 

At the moment we have a workaround for this behaviour using the drop_duplicates() function of pandas.
